common/src/main/scala/org/apache/celeborn/common/util/Utils.scala

```diff
@@ -1282,24 +1282,30 @@ object Utils extends Logging {
 
   def withRetryOnTimeoutOrIOException[T](numRetries: Int, retryWait: Long)(block: => T): T = {
     var retriesLeft = numRetries
+
+    def waitOrThrow(e: Throwable): Unit = {
+      if (retriesLeft > 0) {
+        val retryWaitMs = new Random().nextInt(retryWait.toInt)
+        try {
+          TimeUnit.MILLISECONDS.sleep(retryWaitMs)
+        } catch {
+          case _: InterruptedException =>
+            throw e
+        }
+      } else {
+        throw e
+      }
+    }
+
     while (retriesLeft >= 0) {
       retriesLeft -= 1
       try {
         return block
       } catch {
+        case e: CelebornIOException if e.getCause != null && e.isInstanceOf[IOException] =>
```

Comment:
https://github.com/apache/celeborn/blob/60fa6d0ee786248b9f69c5c937f727884a4a3768/common/src/main/scala/org/apache/celeborn/common/exception/CelebornIOException.scala#L24-L25

CelebornIOException is class of IOException and should be handled below.

Reply:
You shall mean CelebornException.

It is better that you can provide a exception example in the PR description.